metadata-ingestion/src/datahub/ingestion/source/sql/oracle.py

```diff
@@ -31,6 +33,9 @@
 
 logger = logging.getLogger(__name__)
 
+oracledb.version = "8.3.0"
+sys.modules["cx_Oracle"] = oracledb
```

Comment:
why do we need this? this feels a bit scary

can we not just replace all usages of cx_oracle with oracledb?

Reply:
SQLAlchemy 1.4.0 has native support for cx_oracle with 2.0 having native support for oracledb. In 1.4.0 you need this to override the native selection of cx_oracle. When SQLAlchemy is bumped to v2.0.0 this can be removed. Example issue that arises without this line [here](https://stackoverflow.com/questions/74093231/nosuchmoduleerror-cant-load-plugin-sqlalchemy-dialectsoracle-oracledb).

Reply:
As mentioned in the [oracledb](https://python-oracledb.readthedocs.io/en/latest/user_guide/appendix_c.html#python-frameworks-sql-generators-and-orms) docs this code is advised with old versions of SQLAlchemy

Reply:
Both links suggest that the scheme needs to be `oracle` for Sqlalchemy 1.4 and `oracle+oracledb` for Sqlalchemy 2.0.

Without this, oracle ingestion does not work with default scheme, as sqlalchemy is pinned to version less than 2.0 in our dependencies. Error `<class 'sqlalchemy.exc.NoSuchModuleError'>: Can't load plugin: sqlalchemy.dialects:oracle.oracledb`

Fixed in https://github.com/datahub-project/datahub/pull/11829